filebeat/crawler/registrar.go

```diff
@@ -74,7 +77,7 @@ func (r *Registrar) LoadState() {
 		defer existing.Close()
 		logp.Info("Loading registrar data from %s", r.registryFile)
 		decoder := json.NewDecoder(existing)
-		decoder.Decode(&r.State)
+		decoder.Decode(r.getFullState())
```

Comment:
with r.getFullState being a copy, where do you store the results of decoder.Decode?